placeSpecificContent/hq/courtyard/controlPanel/controlPanel_server.js

```diff
@@ -0,0 +1,406 @@
+                        backgroundColor: BUTTON_TRANSITION_COLOR,
+                        textColor: BUTTON_INACTIVE_COLOR
+                    });
+                } else {
+                    Entities.editEntity(button, {
+                        backgroundColor: BUTTON_INACTIVE_COLOR,
+                        textColor: BUTTON_TRANSITION_COLOR
+                    });
+                }
+            });
+
+            var soundPosition = Entities.getEntityProperties(courtyard, 'position').position;
+            _this.playSound(TRANSITION_SOUND, TRANSITION_VOLUME, soundPosition, false, true);
+            
+            if (!roundtableInPlace && roundtableVelocity) {
+                Entities.editEntity(roundtable, {
+                    velocity: { x: 0, y: roundtableVelocity, z: 0 }
+                });
+            }
+
+            if (!courtyardInPlace && courtyardVelocity) {
+                Entities.editEntity(courtyard, {
+                    velocity: { x: 0, y: courtyardVelocity, z: 0 }
+                });
+            }
+
+            if (checkInterval) {
+                Script.clearInterval(checkInterval);
+                checkInterval = null;
+            }
+
+            checkInterval = Script.setInterval(function() {
+                _this.checkEntityPositions(buttonID, targetCourtyardPosition, targetRoundtablePosition);
+            }, CHECK_INTERVAL_MS);
+        },
+
+        /* Check each moveable entity and if it is still moving and close enough to its target position, slam it and clear 
+        its velocity. If both entities are in place, play the done sound and set the button color to active and stop the 
+        position checking interval.
+        */
+        checkEntityPositions: function(buttonID, targetCourtyardPosition, targetRoundtablePosition) {
+            if (!courtyardInPlace) {
+                var courtyardCurrentPosition = Entities.getEntityProperties(courtyard, 'position').position;
+                var courtyardDistanceToTarget = courtyardCurrentPosition.y.toFixed(FIXED_DECIMAL_PLACES) - 
+                targetCourtyardPosition.y.toFixed(FIXED_DECIMAL_PLACES);
+                if ((courtyardVelocity < 0 && courtyardDistanceToTarget <= 0) || 
+                    (courtyardVelocity > 0 && courtyardDistanceToTarget >= 0)) {
+                    courtyardInPlace = true;
+                }
+                if (courtyardInPlace) {
+                    courtyardVelocity = { x: 0, y: 0, z: 0 };
+                    Entities.editEntity(courtyard, {
+                        velocity: { x: 0, y: 0, z: 0 },
+                        position: targetCourtyardPosition
+                    });
+                }
+            }
+
+            if (!roundtableInPlace){
+                var roundtableCurrentPosition = Entities.getEntityProperties(roundtable, 'position').position;
+                var roundtableDistanceToTarget = roundtableCurrentPosition.y.toFixed(FIXED_DECIMAL_PLACES) - 
```

Comment:
- Why are you using `toFixed` here?
- Why not use `Vec3.distance()` here?
- This isn't a distance, since it could be negative. Did you intend for this to be a distance? If so, you'll need to wrap this in `Math.abs()` (but only if you're not going to use `Vec3.distance()`)

Reply:
https://github.com/highfidelity/hifi-content/pull/462#discussion_r325268910